zerver/models.py

```diff
@@ -785,7 +787,8 @@ class Meta:
 def receives_offline_notifications(user_profile):
     # type: (UserProfile) -> bool
     return ((user_profile.enable_offline_email_notifications or
-             user_profile.enable_offline_push_notifications) and
+             user_profile.enable_offline_push_notifications or
+             user_profile.enable_stream_email_notifications) and
```

Comment:
Should this setting go in receives_offline_notifications? If so, should the stream push notifications setting also show up here?

Reply:
I think `receives_stream_notifications` is meant to handle that case?

I guess it depends whether we think `enable_offline_email_notiifcations` should be a requirement of setting the other setting.  I think ultimately we're going to want to refactor this function to have separate functions for emails/push, since if you look at how they're used, it's independent for email vs push.